src/profile/main.mo

```diff
@@ -6,195 +6,26 @@
  * Stability  : Experimental
  */
 
-import Array "mo:stdlib/array.mo";
-import Hash "mo:stdlib/hash.mo";
-import Iter "mo:stdlib/iter.mo";
-import Result "mo:stdlib/result.mo";
-import Trie "mo:stdlib/trie.mo";
-import Util "../common/util.mo";
+import Directory "./directory.mo";
+import Types "./types.mo";
```

Comment:
I found that have the context object and using the front-end for identity management simplified the canister code quite a bit.